server/src/main/java/org/opensearch/indices/recovery/RemoteRecoveryTargetHandler.java

```diff
@@ -192,16 +192,17 @@ public void indexTranslogOperations(
      *
      * This function is used to force a sync target primary node with source (old primary). This is to avoid segment files
      * conflict with replicas when target is promoted as primary.
-     * @param listener segment replication event listener
      */
     @Override
-    public void forceSegmentFileSync(ActionListener<Void> listener) {
-        final String action = SegmentReplicationTargetService.Actions.FORCE_SYNC;
+    public void forceSegmentFileSync() {
         final long requestSeqNo = requestSeqNoGenerator.getAndIncrement();
-        final ForceSyncRequest request = new ForceSyncRequest(requestSeqNo, recoveryId, shardId);
-        final Writeable.Reader<TransportResponse.Empty> reader = in -> TransportResponse.Empty.INSTANCE;
-        final ActionListener<TransportResponse.Empty> responseListener = ActionListener.map(listener, r -> null);
-        retryableTransportClient.executeRetryableAction(action, request, responseListener, reader);
+        transportService.submitRequest(
+            targetNode,
+            SegmentReplicationTargetService.Actions.FORCE_SYNC,
+            new ForceSyncRequest(requestSeqNo, recoveryId, shardId),
+            TransportRequestOptions.builder().withTimeout(recoverySettings.internalActionLongTimeout()).build(),
```

Comment:
How would a segrep failure be treated vs a network failure?  Are we able to resume recovery after completing phase 1 or would the shard get wiped?

Reply:
A segment replication failure will result in failure of recovery process with shard. So, in event of any exception other than connect exception (mentioned next), the recovery process will be marked failed. I think it is worthwhile to revisit different segrep n/w timeout settings and ensure they are appropriate; without which there are chances of continuous recovery failure due to segment replication. 

On network failure `ConnectTransportException` target sends [REESTABLISH_RECOVERY](https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/indices/recovery/PeerRecoveryTargetService.java#L725) request to source which resumes the recovery process using existing handler. If no such handler is found on source, a `PeerRecoveryNotFound` exception is sent which `retry` the recovery from beginning (creates a new recovery request and copying existing recoveryTarget). `Retry` also happens due DelayRecoveryException (source not owner of shard) or IllegalState of shard (IndexNotFoundException/ShardNotFoundException/IllegalIndexShardStateException).